docs(*): update README.md; add example; use manifest v2

vdice opened the pull request:
- Updates the README.md
- Migrates spin.toml, snippets, etc to manifest v2
- Adds a ~~"rust-classic"~~ rust-standalone example (better name?)

vdice commented on README.md:
```diff
@@ -3,21 +3,71 @@
+### Building from source
+
+This component is now fully componentized and can be used anywhere [WebAssembly Components](https://component-model.bytecodealliance.org/)
+are supported.
```
Bold claims are often from those less knowledgeable.  Accurate?  Reword?

dicej replied:
I would say it can be used anywhere `wasi:http@0.2.0-rc-2023-10-18` WebAssembly Components are supported.  We'll be able to upgrade that to a much more meaningful claim once WASI Preview 2 is released and widely adopted.

itowlson commented on README.md:
```diff
@@ -3,21 +3,71 @@
+Below we'll look at running the file server directly as well as using component
+composition to integrate this component in with your application logic.
+
+#### Running the file server
```
super nit: this is a lot of heading levels - shall we bust everything up a level or two?

itowlson commented on examples/rust-classic/README.md:
```diff
@@ -0,0 +1,43 @@
+# Rust fileserver example (classic)
```
the computer industry has a horrible tradition of using "classic" to mean "deprecated" - maybe "standalone", unless composing it really is how we expect this to be used in future

itowlson commented on examples/rust-classic/Cargo.toml:
```diff
@@ -0,0 +1,15 @@
+anyhow = "1"
+http = "0.2.9"
+serde = { version = "1.0", features = ["derive"] }
+spin-sdk = { git = "https://github.com/fermyon/spin" }
```
tag?

itowlson commented on examples/rust-classic/Cargo.toml:
```diff
@@ -0,0 +1,15 @@
+[dependencies]
+anyhow = "1"
+http = "0.2.9"
+serde = { version = "1.0", features = ["derive"] }
```
I don't think this is used?

dicej reviewed (approved):
LGTM.  I would maybe mention that Spin 2.0 (or some other runtime that supports `wasi:http@0.2.0-rc-2023-10-18`, such as `wasmtime serve` or NGNIX Unit) is required not just to try the examples but to use this component in any meaningful way.

vdice commented:
Addressed feedback in https://github.com/fermyon/spin-fileserver/pull/45/commits/2b7da52b70b7a25e43eb0358952a76fbab97162b.  Also added a section mentioning its use via the corresponding spin template.
